File: src/main/java/dbconnection/controller/DbController.java
package dbconnection.controller;

import dbconnection.model.ContactTest;
import dbconnection.model.TableContact;
import dbconnection.model.TableCustomer;
import dbconnection.service.impl.ContactServiceImpl;
import dbconnection.service.impl.CustomerServiceImpl;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Controller;
import org.springframework.web.bind.annotation.*;
import org.springframework.web.servlet.ModelAndView;

/**
 * Controller represents bean, which can be called by web.
 * This example contains ContactService's methods calls.
 */

@Controller
@RequestMapping("/testDb")
public class DbController {

    @Autowired
    ContactServiceImpl contactService;

    @Autowired
    CustomerServiceImpl customerService;

    @RequestMapping(path = "/person/{id}", method = RequestMethod.GET)
    public ModelAndView getHello(@PathVariable Long id) {
        ContactTest contactCrm = contactService.getById(id);
        ModelAndView modelAndView = new ModelAndView("index");
        modelAndView.addObject("person", contactCrm);
        return modelAndView;
    }

    @GetMapping(path = "/getAllContacts")
    public @ResponseBody Iterable<ContactTest> getAllContacts() {
        return contactService.getAll();
    }

    @GetMapping(path = "/getContactById")
    public @ResponseBody
    ContactTest getContactById(Long id) {
        return contactService.getById(id);
    }

    @GetMapping(path = "/getContactByFirstName")
    public @ResponseBody
    ContactTest getContactByFirstName(String firstName) {
        return contactService.getByFirstName(firstName);
    }

    @GetMapping(path = "/createContact", params = {"lastName", "firstName"})
    public @ResponseBody ContactTest
    createNewContact(String lastName, String firstName) {
        ContactTest contactTest = new ContactTest();
        contactTest.setFirstName(firstName);
        contactTest.setLastName(lastName);
        return contactService.addContact(contactTest);
    }

    @GetMapping(path = "/deleteById")
    public @ResponseBody void deleteContactById (Long id){
        contactService.delete(id);
    }

    @GetMapping(path = "/deleteAll")
    public @ResponseBody void deleteAll() {
        contactService.deleteAll();
    }

    @GetMapping(path = "/getCrmContactById")
    public @ResponseBody TableContact getCrmContactById(Long id) {
        return contactService.getCrmContactById(id);
    }

    @GetMapping(path = "/getCustomerByObjid")
    public @ResponseBody TableCustomer getCrmCustomerByObjid(Long id) {
        return customerService.getCustomerByObjid(id);
    }
}

File: src/main/java/connectionKotlin/db/controller/DbController.kt
package connectionKotlin.db.controller

import connectionKotlin.db.model.ContactTest
import connectionKotlin.db.service.ContactService
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Controller
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestMapping

@Controller
@RequestMapping(path = ["/db"])
class DbController (@Autowired private val contactService: ContactService) {

    @GetMapping(path = ["/getAllContacts"])
    fun getAllContacts(): List<ContactTest>? {
        return contactService.getAllContactTest()
    }

    @GetMapping(path = ["/getContactTestById"])
    fun getContactTestById(id: Long): ContactTest? {
        return contactService.getContactTestById(id)
    }

    @GetMapping(path = ["/getContactTestByFirstName"])
    fun getContactTestByFirstName(firstName: String): List<ContactTest>? {
        return contactService.getContactTestByFirstName(firstName)
    }

    @GetMapping(path = ["/createContactTest"])
    fun createContactTest(firstName: String, lastName: String): ContactTest {
        val contact = ContactTest()
        contact.firstName = firstName
        contact.lastName = lastName
        return contactService.createContactTest(contact)
    }

    @GetMapping(path = ["/deleteContactTestById"])
    fun deleteContactTestById(id: Long) {
       contactService.deleteContactTestById(id)
    }

    @GetMapping(path = ["/deleteAllContactTest"])
    fun deleteAllContactTest() {
        contactService.deleteAllContactTest()
    }
}